website/docs: adds information about known device logins (#17175)
goauthentik/authentik: website/docs/add-secure-apps/flows-stages/stages/user_login/index.md

## User login stage configuration options

> When creating or editing this stage, you can set the following configuration options.

Author: I don't think that this is strictly necessary. But I see your point so I've added it in

### Session duration

> By default, the authentik session expires when you close your browser (_seconds=0_).

Reviewer: Because the Head3 is "Session settings", we need to leave the actual name of the setting, even if is seems obvious/redundant.

Reviewer: Wait a minute, I am rethinking this now that I scroll down and see the other other settings ... yes, ok I agree that using the H3 to specify the setting name works. Though this would be inconsistent with other docs, were we bullet lists to show UI fields/settings....

## User login stage configuration options

> When creating or editing this stage in the Admin interface, you can set the following configuration options.

Reviewer: ```suggestion
When creating or editing this stage in the Admin interface, you can define configuration options using the following fields.
```

    - Days
    - Weeks

All values accept floating-point values.

### Stay signed in offset

When this is set to a higher value than the default _seconds=0_, the user logging in is shown a prompt, allowing the user to choose if their session should be extended or not. The same syntax as for _Session duration_ applies.

    ![](./stay_signed_in.png)

### Remember device

If set to a duration above 0, a cookie will be stored for the duration specified which will allow authentik to know whether the user is signing in from a known or unknown (new) device.

If there's an existing authenticated user session for the user with the same IP address, authentik also classifies this as a known device. See [here](../../../../sys-mgmt/events/notification_rule_expression_policies.mdx#trigger-alert-when-user-logs-in-from-unknown-device) for an example of how an alert can be configured for logins from unknown devices.

### Network binding and GeoIP binding

When configured, all sessions authenticated by this stage will be bound to the selected network and/or GeoIP criteria.

> Sessions that break this binding will be terminated on use. The created [`logout`](../../../../sys-mgmt/events/event-actions#logout) event will contain additional data related to what caused the binding to be broken:

Reviewer: on use.. use of what, exactly? Use of the binding? Does this mean that if the selected GeoIP binding, for example, is for Continent and Country, but the session is started in the "wrong" continent, then the session is terminated as soon as the binding is hit/used/implemented?

If Yes, let's drop the "in use" and just say "Sessions that break this binding will be terminated."

```json
{
    "asn": {
        "asn": 6805,
        "as_org": "Telefonica Germany",
        "network": "5.4.0.0/14"
    },
    "geo": {
        "lat": 51.2993,
        "city": "",
        "long": 9.491,
        "country": "DE",
        "continent": "EU"
    },
    "binding": {
        "reason": "network.missing",
        "new_value": {
            "asn": 6805,
            "as_org": "Telefonica Germany",
            "network": "5.4.0.0/14"
        },
        "previous_value": {}
    },
    "ip": {
        "previous": "1.2.3.4",
        "new": "5.6.7.8"
    },
    "http_request": {
        "args": {},
        "path": "/if/admin/",
        "method": "GET",
        "user_agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36"
    },
    "logout_reason": "Session binding broken"
}
```

### Terminate other sessions

> When enabled, previous sessions of the user logging in will be revoked. This has no affect on OAuth refresh tokens.

Reviewer: ```suggestion
When enabled, previous sessions of the same user are revoked. This has no affect on OAuth refresh tokens.
```

    - Days
    - Weeks

All values accept floating-point values.

#### Stay signed in offset

> When this is set to a higher value than the default _seconds=0_, the user logging in is shown a prompt, allowing the user to choose if their session should be extended or not. The same syntax as for _Session duration_ applies.

Reviewer: ```suggestion
When the value in **Stay signed in offset** is set to a higher value than the default _seconds=0_, the user logging in is shown a prompt, allowing the user to choose if their session should be extended or not. The same syntax as for _Session duration_ applies.
```
